moto/ec2/models.py

```diff
@@ -674,6 +681,16 @@ def create_from_cloudformation_json(
         instance = reservation.instances[0]
         for tag in properties.get("Tags", []):
             instance.add_tag(tag["Key"], tag["Value"])
+
+        # Associating iam instance profile.
+        # TODO: Don't forget to implement replace_iam_instance_profile_association once update_from_cloudformation_json
+        #  for ec2 instance will be implemented.
+        if properties.get("IamInstanceProfile"):
```

Comment:
Can we get a test for the CF part as well?

Reply:
Done